### DISCUSSION TITLE
Text on a coloured background

### DISCUSSION THREAD

**Discussion by leeenglestone** [StereoKit/StereoKit#389] [Q&A] [answered]
I notice a lot of your text examples are just floating/ transparent.

If I wanted text on a coloured background/square what is the best way to achieve that?

I looked at colourscheme on UI Window but that seems too subtle and effects every instance if Window?

Is the other alternative to use plain flat plane or flat rectangle exported from Blender and import it then add UI to it in StereoKit?

Or would you recommend another way? Textures/Mesh/ some other way?

-- Lee

**Comment by maluoi** [comment, marked as answer]
Currently, if a Window doesn't work for you, I'd probably recommend a colored quad or cube! There's prebuilt primitives that are nice for this in the `Mesh` class, check out [`Mesh.Quad`](https://stereokit.net/Pages/Reference/Mesh.html) and [`Mesh.Cube`](https://stereokit.net/Pages/Reference/Mesh/Cube.html).

Since these have a dimension of 1 and face forward, you can easily scale them to the right size in the transform Matrix you give them. You can also just draw them directly with [`Material.Default`](https://stereokit.net/Pages/Reference/Material/Default.html), or [`Material.Unlit`](https://stereokit.net/Pages/Reference/Material/Unlit.html), and provide the tint color with your call to `Draw` :)

Theming updates in the future will allow for coloring different UI windows separately, but not quite there yet!
